components/renku_data_services/connected_services/utils.py

```diff
@@ -2,9 +2,41 @@
 
 import base64
 import random
+from enum import StrEnum
+
+from renku_data_services.app_config import logging
+from renku_data_services.connected_services.apispec import Provider
+from renku_data_services.connected_services.apispec import ProviderKind as ApiProviderKind
+from renku_data_services.connected_services.models import OAuth2Client, ProviderKind
+from renku_data_services.connected_services.orm import OAuth2ClientORM
+
+logger = logging.getLogger(__name__)
 
 
 def generate_code_verifier(size: int = 48) -> str:
     """Returns a randomly generated code for use in PKCE."""
     rand = random.SystemRandom()
     return base64.b64encode(rand.randbytes(size)).decode()
+
+
+class GitHubProviderType(StrEnum):
```

Comment:
Should we add a TODO here that says "TODO: add a new type of integration 'github_oauth' instead of running this logic".

This would remove the need for `get_github_provider_type` and instead just check the integration type.

Reply:
I added this todo to the function. Hope that is fine. I'd like to do the mentioned change later.